src/huggingface_hub/inference/_common.py

```diff
@@ -259,6 +256,50 @@ def _bytes_to_image(content: bytes) -> "Image":
     return Image.open(io.BytesIO(content))
 
 
+## PAYLOAD UTILS
+
+
+def _prepare_payload(
+    inputs: Union[str, Dict[str, Any], ContentT],
+    parameters: Optional[Dict[str, Any]],
+    expect_binary: bool = False,
+) -> Dict[str, Any]:
+    """
+    Used in `InferenceClient` and `AsyncInferenceClient` to prepare the payload for an API request, handling various input types and parameters.
+    `expect_binary` is set to `True` when the inputs are a binary object or a local path or URL. This is the case for image and audio inputs.
+    """
+    if parameters is None:
+        parameters = {}
+    parameters = {k: v for k, v in parameters.items() if v is not None}
+    has_parameters = len(parameters) > 0
+
+    is_binary = isinstance(inputs, (bytes, Path))
+    # If expect_binary is True, inputs must be a binary object or a local path or a URL.
+    if expect_binary and not is_binary and not isinstance(inputs, str):
+        raise ValueError("Expected binary inputs or a local path or a URL.")
+    # Send inputs as raw content when no parameters are provided
+    if expect_binary and not has_parameters:
+        return {"data": inputs}
+    # If expect_binary is False, inputs must not be a binary object.
+    if not expect_binary and is_binary:
+        raise ValueError("Unexpected binary inputs.")
+
+    json: Dict[str, Any] = {}
+    # If inputs is a dict, update the json payload with its content
+    if isinstance(inputs, dict):
+        json.update(inputs)
```

Comment:
I'm surprised this is needed. I saw that `question_answering` uses it (and that it was already the case before this PR):

```py
        inputs: Dict[str, Any] = {"question": question, "context": context}
        payload = _prepare_payload(inputs, parameters=parameters)
```

But according [to the specs](https://github.com/huggingface/huggingface.js/blob/4b8aa177c4931e8d0f3e1958eecc640285ff5b23/packages/tasks/src/tasks/question-answering/spec/input.json#L8) and to what I would expect as well, the payload should be 

```js
{
    "inputs": {
        "question": ...,
        "context": ...,
    },
    parameters: { ... }
}
``` 

So I think there was a bug in the implementation before this PR and we should fix it in this one. Could you check how things are supposed to behave on a real example? If `inputs` is always needed -which I think is the case- that would simplify things.

Reply:
**EDIT:** I checked in https://huggingface.co/deepset/roberta-base-squad2 inference widget and it's indeed the case:

![image](https://github.com/user-attachments/assets/a8d2fb5d-c961-4f2b-b077-5765c3830aa6)

Reply:
I tried sending this payload (without `inputs`) and it worked. I fixed it though to align with the specs.
```python
{
    "question": "What's my name?",
    "context": "My name is Clara and I live in Berkeley.",
    "parameters": {"top_k": 3},
}
```